### PR TITLE
Rearrange for module support

### PR DESCRIPTION
**Pull request by MikeBishop**
Restructuring several things to enable installation as a module and eventual upload to PyPi.  The individual scripts can still be executed directly and produce the same result, but the package can also be built/installed as a module and invoked with `python3 -m archive-repo archive ...` or `python3 -m archive-repo pr_branches ...` (note `_` instead of `-` because Python doesn't like `-` in submodule names).

The functionality can also be invoked directly from other programs, though it's possible I'll need to change "archive-repo" to a single token for the same `-` issue.  Because paths have changed, the tests will break when I tag a new version; that should be easily fixed, though.

I want to poke at this some more next week, but I'm off the rest of this week and I want a checkpoint before then.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[build-system]
+requires = [
+    "setuptools>=42",
+    "wheel"
+]
+build-backend = "setuptools.build_meta"
```
Comment (martinthomson) [inline pyproject.toml:6]: No EOL here.  I notice that vscode is bad about that, is that the culprit?

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    classifiers=[
+        "Programming Language :: Python :: 3",
+        "License :: OSI Approved :: MIT License",
+        "Operating System :: OS Independent",
+        "Intended Audience :: Information Technology",
+    ],
+    packages=setuptools.find_packages(),
+    python_requires=">=3.6",
+    install_requires=["python-dateutil", "requests"]
+)
```
Comment (martinthomson) [inline setup.py:27]: EOL

**Review by martinthomson — APPROVED** [review]
Very nice.

File: pyproject.toml (hunk 1 above)
Comment (MikeBishop) [reply pyproject.toml]: It turns out that there is an option to automatically insert a newline, but it's off by default.  I've enabled it, so hopefully better now.

**Pull request merged**